submissions/Nik3264/Frogger-Game/app.js

```diff
@@ -0,0 +1,200 @@
+  constructor({ x, y, speed, game, player }) {
+    super(x, y, game);
+    this.sprite = "images/enemy-bug.png";
+    this.speed = speed;
+    this.player = player;
+  }
+
+  update(dt) {
+    this.x +=
+      dt *
+      (this.speed + 0.4 * this.speed * this.game.level) *
+      !this.game.isStop;
+    if (this.x > this.game.frontierForEnemiesRight) {
+      this.x = this.game.frontierForEnemiesLeft;
+    }
+    if (this.isCollision()) {
+      this.player.start();
+      this.game.level = 1;
+    }
+  }
+
+  isCollision() {
+    return (
+      this.x > this.player.x - this.game.collisionX &&
+      this.x < this.player.x + this.game.collisionX &&
+      this.y > this.player.y - this.game.collisionY &&
+      this.y < this.player.y + this.game.collisionY
+    );
+  }
+
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+}
+
+class Player extends Character {
+  constructor(game) {
+    super(1, 1, game);
+    this.sprite = "images/char-boy.png";
+    this.x = this.game.PLAYER_X_START;
+    this.y = this.game.PLAYER_Y_START;
+  }
+
+  handleInput(key) {
+    switch (key) {
+      case "up":
+        if (this.y > this.game.up) {
+          this.y -= this.game.CELL_HEIGHT;
+        }
+        break;
+      case "down":
+        if (this.y < this.game.down) {
+          this.y += this.game.CELL_HEIGHT;
+        }
+        break;
+      case "left":
+        if (this.x > this.game.left) {
+          this.x -= this.game.CELL_WIDTH;
+        }
+        break;
+      case "right":
+        if (this.x < this.game.right) {
+          this.x += this.game.CELL_WIDTH;
+        }
+        break;
+      default:
+        break;
+    }
+    if (this.isWin()) {
+      this.renderLevel();
+    }
+  }
+
+  renderLevel() {
+    const body = document.querySelector(".message");
+    let div = `<div class="class__modal">You LEVEL${this.game.level}!</div>`;
+    this.game.isStop = true;
+    this.game.level++;
+    this.start();
+    document.removeEventListener("keyup", playerMove);
+    body.innerHTML = div;
+    setTimeout(() => {
+      this.game.isStop = false;
+      document.addEventListener("keyup", playerMove);
+      body.innerHTML = "";
+    }, 1500);
+  }
+
+  isWin() {
+    return this.y <= 0;
+  }
+
+  update() {}
+  start() {
+    this.x = this.game.PLAYER_X_START;
+    this.y = this.game.PLAYER_Y_START;
+  }
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+}
+
+const game = new Game(gameProperties);
+const player = new Player(game);
+
+const createEnemy = (numberOfEnemies) => {
+  const arrOfEnemies = [];
+  for (let i = 0; i < numberOfEnemies; i++) {
+    arrOfEnemies.push(i);
+  }
+  return arrOfEnemies;
+};
+
+const allEnemies = createEnemy(3).map((number) => {
+  return new Enemy({
+    x: 0,
+    y: game.startYForCreateEnemy + number * game.stepYForCreateEnemy,
+    speed: game.increaseSpeed + number * game.increaseSpeed,
+    game: game,
+    player: player,
```

Comment:
There is legal shortcut to create object properties from variables where key will have the name same as the variable in use.
```
{
  variableName,
}
// ..is equivalent to...
{
  variableName: variableName,
}

```